docs/DOCUMENTATION.md

```diff
@@ -0,0 +1,41 @@
+# Technical Documentation
+
+This documentation is to give a technical overview of Remoting Kafka Plugin. To try a demo of the plugin, please visit our [demo](https://github.com/jenkinsci/remoting-kafka-plugin#how-to-run-demo-of-the-plugin) section.
+
+## Overview
+
+Current versions of Jenkins Remoting are based on the TCP protocol. If it fails, the agent connection and the build fails as well. There are also issues with traffic prioritization and multi-agent communications, which impact Jenkins stability and scalability.
+
+This project aims an update of Remoting and Jenkins in order to add support of a popular message queue/bus technology (Kafka) as a fault-tolerant communication layer in Jenkins.
+
+## Architecture
+
+The below picture gives an overview architecture of the plugin.
+
+![Architecture](remoting-kafka-architecture.png)
+
+All command invocations, RMI, class loading, data streaming between master and agent are now transfered to/from a Kafka broker.
+
+## Shared components (kafka-client-lib)
+
+Shared components to be reused between master and agent, with the core [KafkaClassicCommandTransport](https://github.com/jenkinsci/remoting-kafka-plugin/blob/master/kafka-client-lib/src/main/java/io/jenkins/plugins/remotingkafka/commandtransport/KafkaClassicCommandTransport.java). This is the core of the plugin, an extension of Jenkins [CommandTransport](https://github.com/jenkinsci/remoting/blob/master/src/main/java/hudson/remoting/CommandTransport.java), which supports command transports to/from Kafka.
```

Comment:
Same, avoid absolute links where possible. It will allow to maintain multiple branches later if needed